src/Processors/Formats/Impl/ORCBlockInputFormat.cpp

```diff
@@ -165,19 +151,39 @@ void ORCBlockInputFormat::prepareReader()
         /// so we should recursively count the number of indices we need for this type.
         int indexes_count = countIndicesForType(schema->field(i)->type());
         const auto & name = schema->field(i)->name();
-        if (getPort().getHeader().has(name) || nested_table_names.contains(name))
+        const bool contains_column = std::invoke(
```

Comment:
To be honest, don't see much reasons to use `std::invoke` here. If we add flag in `Nested::getAllTableNames` to get the set of names in lower case, we will be able to make this code much easier, like:
```c++
bool case_insensitive = format_settings.parquet.case_insensitive_column_matching;
if (getPort().getHeader().has(name, case_insensitive)
        || nested_table_names.contains(case_insensitive ? boost::to_lower_copy(name) : name))
{
    ...
}
```

Reply:
`std::invoke` was used to eliminate a lot of scopes with if checks, and it's much cleaner than `continue` IMO. 
But with your suggestion for `Nested::getAllTableNames` it's not needed.